cmd/clusterctl/client/common.go

```diff
@@ -25,22 +25,55 @@ import (
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/repository"
 )
 
-// getComponentsByName is a utility method that returns components for a given provider, targetNamespace, and watchingNamespace.
+// getComponentsByNameAndValue is a utility method that returns components
+// for a given provider, targetNamespace, and watchingNamespace.
+func (c *clusterctlClient) getComponentsByNameAndValue(provider string, providerType clusterctlv1.ProviderType, targetNamespace string, watchingNamespace string) (repository.Components, error) {
```

Comment:
nit: The godoc for this is the same as `getComponentsByName`, can we differentiate them or say something like it is the same as `getComponentsByName` except for variables being filled in.

Reply:
what if the user only looks at one of them? I can change it if you feel strongly about it :D